### PR TITLE
:sparkles: (dashboards) piecharts - sorting and label spacing

### PR DESCRIPTION
**Pull request by MatissJanis**
Two changes:

1. the piechart data will now be sorted (from biggest balance to smallest);
2. slightly increased the spacing between the labels in the piechart (for mobile & dashboard page) - to give it a bit more breathing room and so it wouldn't intersect with the piechart itself

### CODE REVIEW

File: packages/desktop-client/src/components/reports/graphs/DonutGraph.tsx [hunk 1]
```diff
@@ -221,6 +221,12 @@ export function DonutGraph({
 
   const [activeIndex, setActiveIndex] = useState(0);
 
+  // Sort the data in the pie chart
+  const unsortedData = data[splitData];
+  const sortedData = useMemo(() => {
+    return unsortedData.sort((a, b) => a[balanceTypeOp] - b[balanceTypeOp]);
+  }, [unsortedData, balanceTypeOp]);
```
[the remark this answers is not in the document]
Comment (MatissJanis) [reply packages/desktop-client/src/components/reports/graphs/DonutGraph.tsx:228]: Ok, but then `getVal` should also be added as a dependency to the `useMemo` hook. And `balanceTypeOp` should be removed then

File: packages/desktop-client/src/components/reports/graphs/DonutGraph.tsx [hunk 2]
```diff
@@ -221,6 +221,10 @@ export function DonutGraph({
 
   const [activeIndex, setActiveIndex] = useState(0);
 
+  // Sort the data in the pie chart
+  const unsortedData = data[splitData];
+  const sortedData = unsortedData.toSorted((a, b) => getVal(b) - getVal(a));
```
[the remark this answers is not in the document]
Comment (MatissJanis) [reply packages/desktop-client/src/components/reports/graphs/DonutGraph.tsx]: Done https://github.com/actualbudget/actual/pull/3855/commits/8202aa52543f8fca064e3b835a834ab1b39a6ebd

**Comment by youngcw** [comment]
The category text is getting cut off just a bit
image

**Comment by matt-fidd** [comment]
Also losing part of the value in the card, should be 1072
image

**Comment by MatissJanis** [comment]
Ok, did a few extra updates to better scale the widget. It's definitely not _perfect_. But better than before :)

**Review by MikesGlitch — APPROVED** [review]

**Comment by MatissJanis** [comment]
I wasn't planing on merging this during the code freeze as it's not really a bugfix.

**Pull request merged**